test/src/e2e_vm_tests/test_programs/should_pass/language/impure_ifs/src/main.sw

```diff
@@ -0,0 +1,46 @@
+script;
```

Comment:
Nice change! Thanks.

Do you mind also adding a test case with `if let`?

Reply:
> Nice change! Thanks.
> 
> Do you mind also adding a test case with `if let` like the one in the linked issue?

Done. But there's a failure in one of the vec tests. Let me try and debug that.